src/scores_ui.cpp

```diff
@@ -55,126 +150,171 @@ static std::string get_achievements_text( const achievements_tracker &achievemen
-    init_windows( ui );
-
-    const std::vector<std::pair<tab_mode, std::string>> tabs = {
-        { tab_mode::achievements, _( "ACHIEVEMENTS" ) },
-        { tab_mode::conducts, _( "CONDUCTS" ) },
-        { tab_mode::scores, _( "SCORES" ) },
-        { tab_mode::kills, _( "KILLS" ) },
-    };
-
-    ui.on_redraw( [&]( const ui_adaptor & ) {
-        werase( w );
-        draw_tabs( w, tabs, tab );
-        draw_border_below_tabs( w );
-        wnoutrefresh( w );
-
-        view.draw( c_white );
-    } );
+    const int monster_kills = kills_data.monster_kill_count();
+    const int npc_kills = kills_data.npc_kill_count();
 
-    while( true ) {
-        if( new_tab ) {
-            switch( tab ) {
-                case tab_mode::achievements:
-                    view.set_text( get_achievements_text( achievements, false, getmaxx( w ) - 2 ) );
-                    break;
-                case tab_mode::conducts:
-                    view.set_text( get_achievements_text( achievements, true, getmaxx( w ) - 2 ) );
-                    break;
-                case tab_mode::scores:
-                    view.set_text( get_scores_text( stats ) );
-                    break;
-                case tab_mode::kills:
-                    view.set_text( kills.get_kills_text() );
-                    break;
-                case tab_mode::num_tabs:
-                    cata_fatal( "Invalid tab" );
-                    break;
+    if( get_option<bool>( "STATS_THROUGH_KILLS" ) &&
+        ImGui::CollapsingHeader( _( "Stats through kills:" ), ImGuiTreeNodeFlags_DefaultOpen ) ) {
+        const int total_kills = kills_data.total_kill_count();
+        ImGui::TextWrapped( string_format( _( "Total kills: %d" ), total_kills ).c_str() );
+        ImGui::NewLine();
+        ImGui::TextWrapped( string_format( _( "Experience: %d (%d points available)" ),
+                                           get_avatar().kill_xp,
+                                           get_avatar().free_upgrade_points() ).c_str() );
+    }
+
+    if( ImGui::CollapsingHeader( string_format( _( "Monster kills (%d):" ), monster_kills ).c_str(),
+                                 ImGuiTreeNodeFlags_DefaultOpen ) ) {
+        if( monster_kills == 0 ) {
+            ImGui::TextWrapped( "%s", _( "You haven't killed any monsters yet!" ) );
+        } else {
+            for( const auto &entry : kills_data.kills ) {
+                const int num_kills = entry.second;
+                const mtype &m = entry.first.obj();
+                const std::string &name = m.nname();
+                const std::string &symbol = m.sym;
+                nc_color color = m.color;
+
+                ImGui::TextColored( c_light_gray, "%d ", num_kills );
+                ImGui::SameLine( 0, 0 );
+                ImGui::TextColored( color, "%s ", symbol );
+                ImGui::SameLine( 0, 0 );
+                ImGui::TextColored( c_light_gray, "%s", name );
```

Comment:
I recommend making this three columns so that the names and counts can line up with each other. And consider making the symbol monospaced.

Reply:
I did not like how the table looked, but we can refine this later.

Reply:
Yea, we can always refine it later.